notes/stv/20201005-automated-deploy-01.txt

```diff
@@ -0,0 +1,596 @@
+at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:54)
+at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:601)
+at scala.concurrent.BatchingExecutor$class.execute(BatchingExecutor.scala:106)
+at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:599)
+at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:44)
+at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:252)
+at scala.concurrent.Promise$class.tryFailure(Promise.scala:112)
+at scala.concurrent.impl.Promise$DefaultPromise.tryFailure(Promise.scala:157)
+at org.apache.spark.rpc.netty.NettyRpcEnv.org$apache$spark$rpc$netty$NettyRpcEnv$$onFailure$1(NettyRpcEnv.scala:206)
+at org.apache.spark.rpc.netty.NettyRpcEnv$$anon$1.run(NettyRpcEnv.scala:243)
+at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
+at java.util.concurrent.FutureTask.run(FutureTask.java:266)
+at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
+at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
+at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
+at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
+at java.lang.Thread.run(Thread.java:748)
+Caused by: java.util.concurrent.TimeoutException: Cannot receive any reply from zeppelin:42175 in 120 seconds
+... 8 more
+
+
+
+## After some investigation this seems to be a network issue
+## We need to open up some ports between Zeppelin & Master
+
+## Fix that got this working was to enable all ingress traffic to Zeppelin from masters & workers
+
+## After restarting nodes, we run the cell again:
+
+%spark.pyspark
+import random 
+NUM_SAMPLES = 2000000000
+
+def inside(p):
+    x, y = random.random(), random.random()
+    return x*x + y*y < 1
+
+count = sc.parallelize(range(0, NUM_SAMPLES)) \
+             .filter(inside).count()
+print ("Pi is roughly %f" % (4.0 * count / NUM_SAMPLES))
+
+> Pi is roughly 3.141593
+
+
+
+# Fetch one Gaia parquet file and try reading from Zeppelin
+
+sudo yum update
+sudo yum install wget
+wget https://cumulus.openstack.hpc.cam.ac.uk:6780/swift/v1/AUTH_21b4ae3a2ea44bc5a9c14005ed2963af/gaia-dr2-parquet-6-8/part-00125-70392076-8b82-4457-8828-22069e7626e9-c000.snappy.parquet
+hdfs dfs -mkdir /gaia
+hdfs dfs -put part-00125-70392076-8b82-4457-8828-22069e7626e9-c000.snappy.parquet  /gaia
+
+
+
+# In Zeppelin..
+
+%spark.pyspark
+gs_df = sqlContext.read.parquet("/gaia/*.parquet")
+
+# following https://spark.apache.org/docs/latest/sql-data-sources-parquet.html
+gs_df.createOrReplaceTempView("gaia_source")
+sql_df = spark.sql("SELECT floor(source_id / 562949953421312) AS hpx5, COUNT(*) AS n, AVG(pmra) AS avgPmRa, AVG(pmdec) AS avgPmDec FROM gaia_source GROUP BY hpx5")
+sql_df.show()
+
+
+# Success!
```

Comment:
What result did you get ?
Was this just one parquet file, or did you run the test on the whole data set ?

Reply:
Haven't documented the exact output, but it was an output of a number of rows from that dataframe. This was just on one parquet file, I then fetched the full dataset and ran the test which matched the output of the prototype on "dev".